rockcraft/plugins/jlink_plugin.py

```diff
@@ -0,0 +1,123 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The Java runtime plugin."""
+
+from typing import Dict, List, Literal, Set, cast
+
+from craft_parts.plugins import Plugin, PluginProperties
+from overrides import override
+
+
+class JLinkPluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the JLink plugin."""
+
+    plugin: Literal["jlink"] = "jlink"
+    jlink_java_version: int = 21
+    jlink_jars: list[str] = []
+    jlink_dep_slices: list[str] = []
```

Comment:
I'm not sure I like this, as it duplicates `stage-packages`. None of the other plugins stage packages like this, and it's probably better to address this in documentation and wait for the resolution of https://github.com/canonical/craft-parts/issues/833

Reply:
We can not use stage-packages here as the plugin needs to use custom chisel release. 
This can be dropped after Java slices are merged. 

Alternatively, and this is probably a better solution - allow specifying custom chisel-releases for the stage-packages.

Reply:
I have dropped the property and updated doc MR to include part about explicitly setting up dependencies for the runtime.